pwiz_tools/Skyline/FileUI/PeptideSearch/BuildPeptideSearchLibraryControl.cs

```diff
@@ -291,36 +295,55 @@ public bool BuildPeptideSearchLibrary(CancelEventArgs e)
             try
             {
                 builder = ImportPeptideSearch.GetLibBuilder(DocumentContainer.Document, DocumentContainer.DocumentFilePath, cbIncludeAmbiguousMatches.Checked);
+                builder.PreferEmbeddedSpectra = PreferEmbeddedSpectra;
             }
             catch (FileEx.DeleteException de)
             {
                 MessageDlg.ShowException(this, de);
                 return false;
             }
-
-            using (var longWaitDlg = new LongWaitDlg
-            {
-                Text = Resources.BuildPeptideSearchLibraryControl_BuildPeptideSearchLibrary_Building_Peptide_Search_Library,
-                Message = Resources.BuildPeptideSearchLibraryControl_BuildPeptideSearchLibrary_Building_document_library_for_peptide_search_,
-            })
+            
+            while (true)

```

Comment:
Maybe more explanatory code:
bool retry;
do {
retry = false;

Reply:
I admit the while(true) loop is kind of lame, but after converting it I think the do/while(retry) version takes more effort to understand than break/continue. Perhaps just add comments to break/continue lines to make the intent clearer?

Reply:
Now I am thinking maybe extract a function that returns a boolean and make this a one line while loop. Anyway, if you push your changes, I will have a look in Visual Studio and see how it feels.